standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5503,15 +5692,34 @@ public Partition get_partition_with_auth(final String db_name,
   @Override
   @Deprecated
   public List<Partition> get_partitions(final String db_name, final String tbl_name,
-                                        final short max_parts) throws NoSuchObjectException, MetaException {
+      final short max_parts) throws NoSuchObjectException, MetaException {
     return get_partitions(db_name, tbl_name,
         new GetPartitionsArgs.GetPartitionsArgsBuilder().max(max_parts).build());
   }
 
   private List<Partition> get_partitions(final String db_name, final String tbl_name,
     GetPartitionsArgs args) throws NoSuchObjectException, MetaException {
     String[] parsedDbName = parseDbName(db_name, conf);
-    startTableFunction("get_partitions", parsedDbName[CAT_NAME], parsedDbName[DB_NAME], tbl_name);
+    List<Partition> ret = null;
+    Exception ex = null;
+    try {
+      PartitionsRequest req = new PartitionsRequest(parsedDbName[DB_NAME], tbl_name);
+      req.setCatName(parsedDbName[CAT_NAME]);
+      req.setMaxParts((short)args.getMax());
```

Comment:
nit: could the `args` have something else configured to get the partitions? such as the include key

Reply:
Good point. I forget about these new changes. Will include in the latest patch